crates/lib3h/src/gateway/gateway_transport.rs

```diff
@@ -192,65 +63,180 @@ impl<'gateway> P2pGateway<'gateway> {
+                    } else {
+                        unimplemented!();
+                    }
+                };
+                // Check if response is an error
+                if let Err(e) = response {
+                    if let Some(parent_msg) = maybe_parent_msg {
+                        parent_msg.respond(Err(Lib3hError::new(ErrorKind::TransportError(e))))?;
+                    }
+                    return Ok(());
+                };
+                let response = response.unwrap();
+                // Must be a SendMessage response
+                match response {
+                    transport::protocol::RequestToChildResponse::SendMessage { payload: _ } => (),
+                    _ => panic!("received unexpected response type"),
+                };
+                println!("yay? {:?}", response);
+                // Act on response: forward to parent
+                if let Some(parent_msg) = maybe_parent_msg {
+                    parent_msg.respond(Ok(GatewayRequestToChildResponse::Transport(response)))?;
+                }
+                // Done
+                Ok(())
+            }),
+        )
+    }
+}
+
+/// Private internals
+impl P2pGateway {
+    /// Handle Transport request sent to use by our parent
+    pub(crate) fn handle_transport_RequestToChild(
+        &mut self,
+        transport_request: transport::protocol::RequestToChild,
+        parent_request: GatewayToChildMessage,
+    ) -> Lib3hResult<()> {
+        match transport_request {
+            transport::protocol::RequestToChild::Bind { spec: _ } => {
+                // Forward to child transport
+                let _ = self.child_transport_endpoint.as_mut().request(
+                    Lib3hTrace,
+                    transport_request,
+                    Box::new(|_me, response| {
+                        let response = {
+                            match response {
+                                GhostCallbackData::Timeout => {
+                                    parent_request
+                                        .respond(Err(Lib3hError::new_other("timeout")))?;
+                                    return Ok(());
+                                }
+                                GhostCallbackData::Response(response) => response,
+                            }
+                        };
+                        // forward back to parent
+                        parent_request.respond(Ok(GatewayRequestToChildResponse::Transport(
+                            response.unwrap(),
+                        )))?;
+                        Ok(())
+                    }),
+                );
+            }
+            transport::protocol::RequestToChild::SendMessage { uri, payload } => {
+                // uri is actually a dht peerAddress
+                // get actual uri from the inner dht before sending
+                let dht_uri_list = self.address_to_uri(&[&uri.to_string()])?;
+                let dht_uri = &dht_uri_list[0];
```

Comment:
I think we should make this do an actual async request on the dht and send the data in the callback.

We can also stop doing the send to multiple... we decided the added code complexity didn't warrant the potential future network throughput benefits